zchunk: 1.1.6-> 1.1.8 & Add darwin support

nima2007 opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change
Zchunk previously did not compile on macOS, so it had to be disabled in librepo for macOS (#107104). 
With this change, Zchunk will compile on macOS and we will be able to enable Zchunk for librepo.

###### Things done
-Updated version and hash from 1.1.6 to 1.1.8
-Added a patch to use the macOS location of endian if compiling on macOS
-Added argp-standalone as a build input for macOS
-Added a patch to find the arg library if compiling on macOS

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS **via nix-review bc I didn't know how to do it any other way**
   - [X] macOS **On macOS Catalina and Big Sur via nix-build & nix-review. https://pastebin.com/tjxtikkn**
   - [ ] other Linux distributions
- [X] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests)) **Did not find any related tests**
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"` **Ran nixpkgs-review pr successfully but I didn't use that exact command**
- [X] Tested execution of all binary files (usually in `./result/bin/`) **https://pastebin.com/RDQwfGFV**
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date **No documentation**
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md). **Squashed to a single commit following commit message format**

rb2k commented on pkgs/development/libraries/zchunk/default.nix:
```diff
@@ -29,13 +30,16 @@ stdenv.mkDerivation rec {
   buildInputs = [
     zstd
     curl
+    argp-standalone # macOS lacks argp
```
If this is just needed for darwin, you can probably do something like

```
] ++ stdenv.lib.optional stdenv.isDarwin argp-standalone;
```

rb2k commented on pkgs/development/libraries/zchunk/darwin.patch:
```diff
@@ -0,0 +1,26 @@
+diff --git a/meson.build b/meson.build
```
https://github.com/zchunk/zchunk/issues/24 mentions that in theory they have osx. I wonder if we can contribute that.

(would still be required here until they merge)

nima2007 replied:
I am seeing both the endian and argp problems mentioned in that issue, but I don't see a fix in any commits. It looks like someone figured it out without making a pull request. So yes, it would make sense for me to make a pull request there too, and then eventually remove the patch here.

rb2k replied:
I don't think that's a blocker, but I think that would be smart.
Otherwise we'll just have to adapt the patch for future updates when it would be easier to just have it done upstream.

SuperSandro2000 commented:
Result of `nixpkgs-review pr 107443` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>zchunk</li>
  </ul>
</details>

nima2007 commented:
Result of `nixpkgs-review pr 107443` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>libdnf</li>
    <li>librepo</li>
    <li>microdnf</li>
    <li>python37Packages.librepo</li>
    <li>python38Packages.librepo</li>
    <li>python39Packages.librepo</li>
    <li>rpm-ostree</li>
    <li>zchunk</li>
  </ul>
</details>

rb2k commented:
Result of `nixpkgs-review pr 107443` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>zchunk</li>
  </ul>
</details>

SuperSandro2000 commented:
> Result of `nixpkgs-review pr 107443` [1](https://github.com/Mic92/nixpkgs-review)
> 
> 8 packages built:

please update your nixpkgs-review.

rb2k commented:
Result of `nixpkgs-review pr 107443` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>libdnf</li>
    <li>librepo</li>
    <li>microdnf</li>
    <li>python37Packages.librepo</li>
    <li>python38Packages.librepo</li>
    <li>python39Packages.librepo</li>
    <li>rpm-ostree</li>
    <li>zchunk</li>
  </ul>
</details>

nima2007 committed:
zchunk: 1.1.6 -> 1.1.8 | Add darwin support

Zchunk previously couldn't compile on macOS and had to be skipped for macOS on some packages.
